CMakeLists.txt

```diff
@@ -85,12 +89,12 @@ INCLUDE_DIRECTORIES(
   ${LEMON_INCLUDE_DIRS}
   ${CLI11_INCLUDE_DIRS}
   ${LIBQASM_INCLUDE_DIRS}
-#  ${EIGEN_INCLUDE_DIRS}
-#  ${EIGEN_UNSUPPORTED_INCLUDE_DIRS}
+  ${EIGEN_INCLUDE_DIRS}
+  ${EIGEN_UNSUPPORTED_INCLUDE_DIRS}
   ${PROJECT_SOURCE_DIR}/src
 )
-
 IF(CMAKE_COMPILER_IS_GNUCXX)
+  LINK_LIBRARIES("-lX11")
```

Comment:
this should not be here, but in SWIG_LINK_LIBRARIES in swig folder. Please check how to define a X11 library following the example of lemon. Also make cmake search for the library and give a configuration error if not found.